daemon/cmd/daemon_test.go

```diff
@@ -158,13 +151,8 @@ func (ds *DaemonSuite) SetUpTest(c *C) {
 				cs.Disable()
 				return cs
 			},
-			func() datapath.Datapath { return fakeDatapath.NewDatapath() },
-			func(dp datapath.Datapath) datapath.NodeIDHandler { return dp.NodeIDs() },
 			func() *option.DaemonConfig { return option.Config },
-			func() cnicell.CNIConfigManager { return &fakecni.FakeCNIConfigManager{} },
-			func() signalmap.Map { return fakesignalmap.NewFakeSignalMap([][]byte{}, time.Second) },
-			func() authmap.Map { return fakeauthmap.NewFakeAuthMap() },
-			func() egressmap.PolicyMap { return nil },
+			fakeDatapath.Cell,
```

Comment:
In `suite/agent.go` the `fakeDatapath.Cell` is placed directly in the hive definition. Here it appears nested in the `cell.Provide(...)`. I'm not great on my hive-fu so I'm not sure if it makes a difference. If it has to be this way, I'm curious to know why.

Reply:
This most definitely doesn't work. I was sloppy with the changes in this file. Will fix this and the CNIConfigManager one and make sure to integration test locally.